### ISSUE TITLE
Uncommitted changes after git octopus execution (version 2.0)

### ISSUE THREAD

**Issue by danoliv** [lesfurets/git-octopus#34]
Sometimes when we execute git octopus on multiple branches (3 in this example) the merge is apparently successful but some commits are not present in the git log, instead there are the corresponding changes uncommitted in the working copy.

Step to reproduce the bug:

First we prepare the repository:

```bash
mkdir mergebug/
cd mergebug/
git init
touch test
git add test
git commit -a -m "commit test"
git checkout -b qa
git checkout -b feature/test
touch testFeature
git add testFeature
git commit -a -m "add testFeature"
git checkout master
```
Then we execute git octopus to merge the 3 branches:
```
git octopus feature/* qa

2 branches (I)ncluded (2 matching, 0 (E)xcluded):
I  refs/heads/feature/test
I  refs/heads/qa

Fast-forwarding to: refs/heads/feature/test
Already up-to-date with refs/heads/qa
```
We expect to have the "add testFeature" commit in the log but git log will show only the first commit:
```
git log --oneline --decorate

dfc4c37 (HEAD -> master, qa) commit test
```

And git status will show the uncommitted file:
```
git status

On branch master
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	new file:   testFeature
```

**Comment by apflieger** [comment]
Any volunteer to write this as a test case?

**Comment by danoliv** [comment]
We added a test case and fixed the problem. Please check https://github.com/lesfurets/git-octopus/pull/35.

**Comment by hvihvi** [comment]
I tried writing the test case in Go, but the test doesn't fail when it should, haven't figured out why yet.
https://github.com/lesfurets/git-octopus/commit/ded2105af7fc08e44d29a1e2c25dac92c50a11cb

**Comment by apflieger** [comment]
The test was correct, you just did the wrong assertion. 
I fixed it so the test is properly failing now. I simplified and moved it to `basics_test.go` because it really is, basic. 
@danoliv could you rebase on this branch (`test_should_not_pass`) or merge it in your PR?
beside that, your change breaks TestOctopusAlreadyUpToDate :(

**Comment by danoliv** [comment]
I rebased from the `test_should_not_pass` branch. Sorry I launched only the old tests with Docker and not the new ones.

**Comment by danoliv** [comment]
I found out that the fast-forward section in the mergeHeads was introducing the bug, so I reverted my previous modification and removed that section. All tests are ok now.

**Comment by apflieger** [comment]
Fixed by #36
